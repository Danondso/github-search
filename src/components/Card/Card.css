.card-container {
    border-radius: 15px;
    background-color: '#fff';
    width: 100%;
}

.card-content {
    display: flex;
    flex-direction: column;
    align-items: center;
    border-radius: 0 0 15px 15px;
}

.card-header {
    padding: 0.5rem;
    border-radius: 15px 15px 0 0;
    padding-bottom: 1rem;
    font-size: 1.5rem;
}

@media screen and (max-width: 600px) {
    .card-header {
        font-size: 0.75rem;
    }
  }
