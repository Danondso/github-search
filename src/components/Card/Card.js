import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

export function Card({ children }) {
  return (
    <div className="card-container">
      {children}
    </div>
  );
}

export function CardHeader({ headerText, headerStyle }) {
  return (<div className="card-header" style={headerStyle}>{headerText}</div>
  );
}

export function CardContent({ children, contentStyle }) {
  return (
    <div className="card-content" style={contentStyle}>
      {children}
    </div>
  );
}

Card.propTypes = {
  children: PropTypes.node.isRequired,
};

CardHeader.propTypes = {
  headerText: PropTypes.string.isRequired,
  // I prefer not to disable per line unless it's really required,
  // ideally I have a custom type for this but I'm foregoing that.
  // eslint-disable-next-line react/forbid-prop-types
  headerStyle: PropTypes.object,
};

CardHeader.defaultProps = {
  headerStyle: {
    backgroundColor: '#fff',
    color: '#000',
  },
};

CardContent.propTypes = {
  children: PropTypes.node.isRequired,
  // I prefer not to disable per line unless it's really required,
  // ideally I have a custom type for this but I'm foregoing that.
  // eslint-disable-next-line react/forbid-prop-types
  contentStyle: PropTypes.object,
};

CardContent.defaultProps = {
  contentStyle: {
    backgroundColor: '#fff',
  },
};
