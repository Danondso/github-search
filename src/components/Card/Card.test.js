import React from 'react';
import { render, screen } from '@testing-library/react';
import { Card, CardContent, CardHeader } from './Card';

describe('Card Test Suite', () => {
  test('renders card with children', () => {
    render(
      <Card>
        I am text!
      </Card>,
    );
    const card = screen.getByText('I am text!');
    expect(card).toBeInTheDocument();
  });

  test('renders CardHeader with headerText', () => {
    render(
      <Card>
        <CardHeader headerText="Test" />
      </Card>,
    );
    const cardHeader = screen.getByText('Test');
    expect(cardHeader).toBeInTheDocument();
  });

  test('renders CardContent with children', () => {
    render(
      <Card>
        <CardHeader headerText="Test" />
        <CardContent contentStyle={{ color: 'pink' }}>
          <div>I am card content!</div>
        </CardContent>
      </Card>,
    );
    const cardContent = screen.getByText('I am card content!');
    expect(cardContent).toBeInTheDocument();
  });
});
