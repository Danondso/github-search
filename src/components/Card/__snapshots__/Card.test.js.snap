// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Test Suite renders CardContent with children 1`] = `
<div>
  I am card content!
</div>
`;

exports[`Card Test Suite renders CardHeader with headerText 1`] = `
<div
  class="card-header"
  style="background-color: rgb(255, 255, 255); color: rgb(0, 0, 0);"
>
  Test
</div>
`;

exports[`Card Test Suite renders card with children 1`] = `
<div
  class="card-container"
>
  I am text!
</div>
`;
